src/partition/repart.c

```diff
@@ -423,24 +423,20 @@ static void partition_foreignize(Partition *p) {
         p->verity = VERITY_OFF;
 }
 
-static bool partition_exclude(const Partition *p) {
-        assert(p);
-
+static bool partition_type_exclude(GptPartitionType type) {
```

Comment:
hmm, copying the whole structure all the time sounds busywork. either just pass the type uuid here, or pass a pointer to the struct?